dora/src/bytecode/writer.rs

```diff
@@ -973,6 +979,11 @@ impl BytecodeWriter {
         self.emit_values(&values);
     }
 
+    pub fn set_position(&mut self, pos: Position) {
+        let offset = self.code.len() as u32;
```

Comment:
Sounds better to me. Should I add a `unset_position` function? So we could avoid that a position is used for unrelated bytecodes.
Also we have to call the set_position after a sub-expression is visited, because it could set the expression somewhere.

Reply:
I don't think this is necessary. Ideally we would always emit bytecode through some wrapper around BytecodeWriter that ensures that a Position was set.

Reply:
To clarify, you mean the `unset_position` or the call after a sub-expression or both?

Reply:
unset_position

Reply:
OK, I changed `set_position` to set the position in a field. `emit_values` uses this value and binds the position to the offset.
I call `set_position` now at the top of `visit_stmt` and `visit_expr`, additionally before a bytecode is emitted inside a visitor function, if a sub-expression was called previously.